contracts/AutonomousDripper.sol

```diff
@@ -0,0 +1,187 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@chainlink/contracts/src/v0.8/ConfirmedOwner.sol";
+import "@chainlink/contracts/src/v0.8/KeeperCompatible.sol";
+import "@openzeppelin/contracts/finance/VestingWallet.sol";
+import "@openzeppelin/contracts/security/Pausable.sol";
+
+/**
+ * @title The AutonomousDripper Contract
+ * @author gosuto.eth
+ * @notice A Chainlink Keeper-compatible version of OpenZeppelin's
+ * VestingWallet; removing the need to monitor the interval and/or call release
+ * manually. Also adds a (transferable) owner that can set the address of the
+ * keeper's registry, pause/unpause the contract and sweep all ether/ERC-20
+ * tokens.
+ * Takes strong inspiration from https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.8/upkeeps/EthBalanceMonitor.sol
+ */
+contract AutonomousDripper is VestingWallet, KeeperCompatibleInterface, ConfirmedOwner, Pausable {
+    event EtherSwept(uint256 amount);
+    event ERC20Swept(address indexed token, uint256 amount);
+    event KeeperRegistryAddressUpdated(address oldAddress, address newAddress);
+
+    error OnlyKeeperRegistry();
+
+    uint public lastTimestamp;
+    uint public immutable interval;
+    address[] public assetsWatchlist;
+    address private _keeperRegistryAddress;
+
+    constructor(
+        address beneficiaryAddress,
+        uint64 startTimestamp,
+        uint64 durationSeconds,
+        uint intervalSeconds,
+        address[] memory watchlistAddresses,
+        address keeperRegistryAddress
+    ) VestingWallet(
+        beneficiaryAddress,
+        startTimestamp,
+        durationSeconds
+    ) ConfirmedOwner(
+        msg.sender
+    ) {
+        lastTimestamp = startTimestamp;
+        interval = intervalSeconds;
+        assetsWatchlist = watchlistAddresses;
+        setKeeperRegistryAddress(keeperRegistryAddress);
+    }
+
+    /**
+     * @dev Setter for the list of ERC-20 token addresses to consider for
+     * releasing. Can only be called by the current owner.
+     */
+    function setAssetsWatchlist(address[] calldata newAssetsWatchlist) public virtual onlyOwner {
+        assetsWatchlist = newAssetsWatchlist;
+    }
+
+    /**
+     * @dev Loop over the assetsWatchlist and check their local balance.
+     * Returns a filtered version of the assetsWatchlist for which the local
+     * balance is greater than zero.
+     */
+    function _getAssetsHeld() internal view returns (address[] memory) {
+        address[] memory _assetsHeld = new address[](assetsWatchlist.length);
+        uint256 count = 0;
+        for (uint idx = 0; idx < assetsWatchlist.length; idx++) {
+            uint256 balance = IERC20(assetsWatchlist[idx]).balanceOf(address(this));
+            if (balance > 0) {
+                _assetsHeld[count] = assetsWatchlist[idx];
+                count++;
+            }
+        }
+        if (count != assetsWatchlist.length) {
+            assembly {
```

Comment:
What's the deal with random uncommented assembly?

Reply:
I think it comes from here:
https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.8/upkeeps/EthBalanceMonitor.sol#L110-L113
Looks like it truncates the return data array size

Reply:
Would just add comment

Reply:
276fe10b45fa3573c55ded5b12fc33f71bee86d9